sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,182 @@ class RotatingCertificateCredential implements TokenCredential {
+
+```
+import { useIdentityExtension } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+```
+
+After calling `useIdentityExtension`, the persistent token cache extension is registered to the `@azure/identity` package and will be available on all credentials that support persistent token caching (those that have `tokenCachePersistenceOptions` in their constructor options).
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
+
+To configure a credential, such as the `InteractiveBrowserCredential`, to persist token data, simply set the `tokenCachePersistenceOptions` option.
+
+```
+import { useIdentityExtension, InteractiveBrowserCredential, InteractiveBrowserCredentialOptions } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+const options: InteractiveBrowserCredentialOptions =  {
+  tokenCachePersistenceOptions: {
+    enabled: true
+  }
+}
+const credential = new InteractiveBrowserCredential(options);
+```
+
+#### Persisting the Authentication Record
+
+The `AuthenticationRecord` which is returned from the `authenticate`, contains data identifying an authenticated account. It is needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state.
+
+Here is an example of an application storing the `AuthenticationRecord` to the local file system after authenticating the user.
+
+```
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+const AUTH_RECORD_PATH = "./tokencache.bin";
+
+const authRecord: AuthenticationRecord = await credential.authenticate();
+const writeFileAsync = promisify(fs.writeFile);
+const content = serializeAuthenticationRecord(authRecord);
+await writeFileAsync(path.join(process.cwd(), AUTH_RECORD_PATH), content);
```

Comment:
Most Node versions don’t have await at this level. We tend to wrap the code of these samples in an asynchronous function, like `async function main() {`, so that you can call `await` inside of the function without problems.

Reply:
Same applies for all the other cases where you’re using top-level awaits